packages/clay-navigation-bar/src/Hooks.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {removeCollapseHeight, setCollapseHeight} from './Helpers';
+import {useEffect, useState} from 'react';
+
+export function useTransition(
+	visible: Boolean,
+	setVisible: any,
+	contentRef: React.RefObject<any>
+) {
+	const [transitioning, setTransitioning] = useState<Boolean>(false);
+
+	useEffect(() => {
```

Comment:
I wonder if this should be "useLayoutEffect." Any chance you happen to know when to use effect vs layoutEffect, i've been a bit confused by those two.

Reply:
Makes sense changing to useLayoutEffect @bryceosterhaus! `useEffect` can be used in the most of use cases but when you are mutating DOM elements we need use `useLayoutEffect`.

- https://kentcdodds.com/blog/useeffect-vs-uselayouteffect

Reply:
I'm not sure if this should be a `useLayoutEffect`, it will cause a new rendering and will only mutate after it completes the latest renderings... And it can cause a blocking visual updates (which in this case is what we are trying to do here), I'd rather go with `useEffect` if it causes problems we can go to `useLayoutEffect`.